lib/archethic/crypto.ex

```diff
@@ -1168,6 +1168,24 @@ defmodule ArchEthic.Crypto do
   @spec default_curve() :: supported_curve()
   def default_curve, do: Application.get_env(:archethic, __MODULE__)[:default_curve]
 
+  @doc """
+  Get the origin seed for a given origin family
+  """
+  @spec get_origin_family_seed(supported_origin()) :: binary()
+  def get_origin_family_seed(origin_id) do
+    storage_nonce() <>
+      case origin_id do
```

Comment:
I think you can leverage `ID.from_origin` which will return an integer for the given origin family

Reply:
Exemple: 
```elixir
<<storage_nonce()::binary, ID.from_origin(origin_id)::8>>
```

Reply:
Instead of putting this in the crypto.ex, you should move it to shared_secrets.ex, to avoid confusion with origin_id for public keys